File: app/src/main/java/nejati/me/service/generator/ServiceGenerator.java
package nejati.me.service.generator;

import javax.inject.Inject;

import nejati.me.service.api.RetroClient;
import retrofit2.Retrofit;

public class ServiceGenerator {
    private Retrofit retrofit;

    @Inject
    public ServiceGenerator(Retrofit retrofit) {
        this.retrofit = retrofit;
    }

    public RetroClient createService() {
        return retrofit.create(RetroClient.class);
    }


}

File: app/src/main/java/nejati/me/service/generator/ServiceGenerator.kt
package nejati.me.service.generator

import javax.inject.Inject

import nejati.me.service.api.RetroClient
import retrofit2.Retrofit

class ServiceGenerator @Inject
constructor(private val retrofit: Retrofit) {

    fun createService(): RetroClient {
        return retrofit.create(RetroClient::class.java)
    }


}
